### PR TITLE
separate `push` and `push.create` commands

### PR DESCRIPTION
**Pull request by aryairani**
Separate `push` and `push.create` commands. Regular `push` won't create a new namespace at the target and will prompt the user to do `push.create` if they are intending to create a new namespace and it's not a typo.

__Todo:__
- [ ] Move the `empty` check to look at the selected sub-namespace.
- [ ] Update tests to use `push.create` as appropriate (tests are failing due to this currently)
- [ ] Update error-on-empty message to include the path that was empty
- [ ] Add a test that confirms failure when doing `push` to empty namespace
- [ ] Add a test that confirms failure when doing `push.create` to nonempty namespace?

Other random stuff:

* The `help` command was doing `Pretty.wrap` around all the command help, which mangles things badly for commands that have more structured help. It took me like an hour to track this down. Correct behavior is for the commands to use `Pretty.wrap` internally if they want wrapped text.

Screen Shot 2021-05-20 at 11 35 09 PM

Here's the error you get if you try to use `push` and the destination is an empty namespace:

Screen Shot 2021-05-20 at 11 59 15 PM

* This PR was cherry-picked from another branch that will be rebased soon.  Here are the original commits in case they are needed (https://github.com/unisonweb/unison/tree/feature/push-create-archive) which I guess can be deleted once we're satisfied here.

### CODE REVIEW

File: parser-typechecker/src/Unison/Codebase/FileCodebase.hs [hunk 1]
```diff
@@ -292,13 +293,16 @@ pushGitRootBranch
...
   -> Branch m
   -> RemoteRepo
   -> SyncMode
   -> ExceptT GitError m ()
-pushGitRootBranch syncToDirectory cache branch repo syncMode = do
+pushGitRootBranch syncToDirectory cache onEmptyDest branch repo syncMode = do
   -- Pull the remote repo into a staging directory
   (remoteRoot, remotePath) <- viewRemoteBranch' cache (repo, Nothing, Path.empty)
+  when (remoteRoot == Branch.empty && onEmptyDest == AbortOnEmptyDest) $
+    throwError $ GitError.PushDestinationIsEmpty repo
```
Comment (aryairani) [inline parser-typechecker/src/Unison/Codebase/FileCodebase.hs:305]: This is checking if the repo root is empty, but I think we instead want it to check the target sub-namespace before we get in here.

**Comment by aryairani** [comment]
(How do I turn it back into a Draft PR after the fact?)

**Comment by aryairani** [comment]
@pchiusano Maybe it should fail if you say `push.create` and it's _not_ empty, too.

**Comment by ChrisPenner** [comment]
I'm assuming this is closed by https://github.com/unisonweb/unison/pull/2600 ; feel free to clarify and re-open if not 😄
